src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4206,37 +4339,76 @@ public override BoundNode VisitAsOperator(BoundAsOperator node)
 
             //if (this.State.Reachable) // Consider reachability: see https://github.com/dotnet/roslyn/issues/28798
             {
-                bool? isNullable = null;
+                NullableAnnotation nullableAnnotation = NullableAnnotation.Unknown;
                 if (!node.Type.IsValueType)
                 {
                     switch (node.Conversion.Kind)
                     {
                         case ConversionKind.Identity:
+                            // Inherit nullability from the operand
+                            nullableAnnotation = _resultType.NullableAnnotation;
+                            break;
+
                         case ConversionKind.ImplicitReference:
+
                             // Inherit nullability from the operand
```

Comment:
>// Inherit nullability from the operand [](start = 28, length = 39)

the comment seems off

Reply:
> the comment seems off

In what sense? Looks accurate to me, we are inheriting nullability from the operand. 

---
In reply to: [236482286](https://github.com/dotnet/roslyn/pull/31264#discussion_r236482286) [](ancestors = 236482286)

Reply:
I can see how it made sense when the logic was just `nullableAnnotation = _resultType.NullableAnnotation;`. Now it's only correct in a loose sense.
If we use it in original/narrow sense, then it should be removed here. If we use it in loose sense, then it applies to all cases (including `ConversionKind.Boxing`) and can be moved out or removed.

---
In reply to: [236482834](https://github.com/dotnet/roslyn/pull/31264#discussion_r236482834) [](ancestors = 236482834,236482286)

Reply:
Inherit is not the same as copy. The comment is accurately describes what the code is doing. 

---
In reply to: [236492967](https://github.com/dotnet/roslyn/pull/31264#discussion_r236492967) [](ancestors = 236492967,236482834,236482286)

Reply:
Does the same comment apply to all the cases, such as the the Boxing case, or not?

---
In reply to: [236779328](https://github.com/dotnet/roslyn/pull/31264#discussion_r236779328) [](ancestors = 236779328,236492967,236482834,236482286)